src/docs/development/platform-integration/add-to-app/performance.md

```diff
@@ -0,0 +1,174 @@
+
+On **iOS**, this happens when you call [`runWithEntrypoint:`]({{site.api}}/objcdoc/Classes/FlutterEngine.html#/c:objc(cs)FlutterEngine(im)runWithEntrypoint:)
+on a `FlutterEngine`.
+
+At this point, your Dart code's selected entrypoint (the `main()` function of
+your Dart library's main.dart file by default) is executed. If you called the
+Flutter function [runApp()]({{site.api}}/flutter/widgets/runApp.html) in your
+`main()` function, your Flutter app or library's widget tree will also be
+created and built. If you need to gate certain functionalities from executing
+in your Flutter code, the `AppLifecycleState.detached` enum value will indicate
+that the `FlutterEngine` is not yet attached to any UI components such as a
+`FlutterViewController` on iOS or a `FlutterActivity` on Android.
+
+### Attaching UI to the Flutter Engine
+
+A standard full-Flutter app moves to reach this state as soon as the app is
+launched.
+
+In an add-to-app scenario, this happens when you attach a `FlutterEngine`
+to a UI component such as by calling [`startActivity()`](https://developer.android.com/reference/android/content/Context.html#startActivity(android.content.Intent))
+with an [`Intent`](https://developer.android.com/reference/android/content/Intent.html)
+built via [`FlutterActivity.withCachedEngine()`]({{site.api}}/javadoc/io/flutter/embedding/android/FlutterActivity.html#withCachedEngine-java.lang.String-)
+on **Android**. Or by presenting a [`FlutterViewController`]({{site.api}}/objcdoc/Classes/FlutterViewController.html)
+initialized via [`initWithEngine: nibName: bundle:`]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)initWithEngine:nibName:bundle:)
+on **iOS**.
+
+This would also the case if a Flutter UI component was launched without
+pre-warming a `FlutterEngine` such as via [`FlutterActivity.createDefaultIntent()`]({{site.api}}/javadoc/io/flutter/embedding/android/FlutterActivity.html#createDefaultIntent-android.content.Context-)
+on **Android** or via [`FlutterViewController initWithProject: nibName: bundle:`]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)initWithProject:nibName:bundle:)
+on **iOS**. An implicit `FlutterEngine` is created in these cases. ;
+
+Behind the scene, both platform's UI components serves to provide the
+`FlutterEngine` with a rendering surface such as a [Surface](https://developer.android.com/reference/android/view/Surface)
+on **Android** or a [CAEAGLLayer](https://developer.apple.com/documentation/quartzcore/caeagllayer)
+or [CAMetalLayer](https://developer.apple.com/documentation/quartzcore/cametallayer)
+on **iOS**.
+
+At this point, the [Layer]({{site.api}}/flutter/rendering/Layer-class.html)
+tree generated by your Flutter program per frame will be converted into
+OpenGL (or Vulkan or Metal) GPU instructions.
+
+## Memory and latency
+
+Showing a Flutter UI has non-trivial latency cost. This cost can be alleviated
```

Comment:
Just curious. Why do you prefer "showing a Flutter UI" to "displaying a Flutter UI"? Is there some meaningful difference? I just find "displaying" to be clearer.

Reply:
This may be completely unjustified. There's a little tingle in the back of my head that makes me think that displaying is a slightly smaller subset of what it takes to show it. As in, to show you results, I'd have to produce it first whereas displaying is slightly more confined to the broadcasting component of it. 

In this case, the non-trivial cost is really the entire process leading up to Flutter rendering, not just the rendering part of it.